File: problems/2399-check-distances-between-same-letters/solution.java
class Solution {
    public boolean checkDistances(String s, int[] distance) {
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int firstIndex = i;
            int secondIndex = -1;
            for (int j = i + 1; j < s.length(); j++) {
                if (s.charAt(j) == c) {
                    secondIndex = j;
                    break;
                }
            }
            if (secondIndex != -1) {
                int charIndex = c - 'a';
                if (secondIndex - firstIndex - 1 != distance[charIndex]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: problems/2399-check-distances-between-same-letters/solution.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        for i in range(len(s)):
            char_index = ord(s[i]) - ord('a')
            if distance[char_index] != -1:
                first_occurrence = i
                second_occurrence = s.find(s[i], i + 1)
                if second_occurrence == -1:
                    continue

                if second_occurrence - first_occurrence - 1 != distance[char_index]:
                    return False

                distance[char_index] = -1

        return True